packages/plugin-workspace-tools/sources/commands/workspace.ts

```diff
@@ -0,0 +1,79 @@
+import {WorkspaceRequiredError}                            from "@berry/cli";
+import {CommandContext, Configuration, Project, Workspace} from "@berry/core";
+import {structUtils}                                       from "@berry/core";
+import {Command, UsageError}                               from "clipanion";
+
+// eslint-disable-next-line arca/no-default-export
+export default class WorkspaceCommand extends Command<CommandContext> {
+  @Command.String()
+  workspaceName!: string;
+
+  @Command.String()
+  commandName!: string;
+
+  @Command.Proxy()
+  args: Array<string> = [];
+
+  static usage = Command.Usage({
+    category: `Workspace-related commands`,
+    description: `run a command on the specified namespace`,
+    details: `
+      This command will run a given sub-command on a single workspace.
+    `,
+    examples: [
+      [
+        `Add a package to a single workspace`,
+        `yarn workspace components add -D react`,
+      ],
+      [
+        `Run build script on a single workspace`,
+        `yarn workspace components run build`,
+      ],
+    ],
+  });
+
+  @Command.Path(`workspace`)
+  async execute() {
+    const configuration = await Configuration.find(
+      this.context.cwd,
+      this.context.plugins
+    );
+    const {project, workspace: cwdWorkspace} = await Project.find(
+      configuration,
+      this.context.cwd
+    );
+
+    if (!cwdWorkspace) throw new WorkspaceRequiredError(this.context.cwd);
```

Comment:
I'm curious, do you have a prettier extension in your editor that autoformats the code like this (I ask because our formatting is handled by ESLint which doesn't generate this type of code)?

Reply:
I manually press "Format code" but that should do the same. I noticed that there were some inconsistencies. For example formatting with prettier + eslint.autFixOnSave produced different formatting than eslint.autoFixOnSave + prettier. 

In any case I was very confused about when eslint complained about brackets for if blocks.

Reply:
I see - we don't actually use Prettier in this repository - I'm experimenting with a more relaxed set of autofixed ESLint rules rather than the blanket formatting Prettier brings, so just running ESLint should be fine.

We'll see whether it scales 😄

Reply:
I see - we don't actually use Prettier in this repository - I'm experimenting with a more relaxed set of autofixed ESLint rules rather than the blanket formatting Prettier brings, so just running ESLint should be fine.

We'll see whether it scales - maybe I'll try adding a prettierignore covering all the files to prevent this particular issue 😄

Reply:
Moved it to the next line.